src/api/services/site-service.ts

```diff
@@ -0,0 +1,78 @@
+import {Injectable} from '@angular/core';
+import {ApiRoot} from "../persistence/ApiRoot";
+import {CoreWebService} from "./core-web-service";
+import {Observable} from 'rxjs/Rx';
+import {RequestMethod} from '@angular/http';
+import {Observer} from "rxjs/Observer";
+
+@Injectable()
+export class SiteService  {
+
+    private allSiteUrl:string;
+    private switchSiteUrl:string;
+
+    private currentSite:Site;
+    private sites:Site[];
+
+    private switchSiteObservable:Observable<Site>;
+    private switchSiteObserver:Observer<Site>;
+
+    constructor(_apiRoot: ApiRoot, private coreWebService: CoreWebService) {
+        this.allSiteUrl = `${_apiRoot.baseUrl}api/v1/site/currentSite`;
+        this.switchSiteUrl = `${_apiRoot.baseUrl}api/v1/site/switch`;
+
+        this.switchSiteObservable = Observable.create( observer => {
+            this.switchSiteObserver = observer;
+
+            if ( this.currentSite ) {
+                this.switchSiteObserver.next(this.currentSite);
+            }
+        });
+    }
+
+    public getAllSites(): Observable<{currentSite:Site, sites:Site[]}> {
```

Comment:
I don't understand why we don't use the ```request``` method we have in CoreWebService for this... if is not working as it is, can we update it to work in this scenario too?